internal/kafka/internal/workers/clusters_mgr_test.go

```diff
@@ -294,19 +272,18 @@ func TestClusterManager_processMetrics(t *testing.T) {
 		},
 	}
 
-	RegisterTestingT(t)
-
 	for _, testcase := range tests {
 		tt := testcase
 		t.Run(tt.name, func(t *testing.T) {
+			g := NewWithT(t)
 			c := &ClusterManager{
 				ClusterManagerOptions: ClusterManagerOptions{
 					ClusterService:         tt.fields.clusterService,
 					DataplaneClusterConfig: tt.fields.dataplaneClusterConfig,
 					SupportedProviders:     tt.fields.supportedProviders,
 				},
 			}
-			Expect(len(c.processMetrics()) > 0).To(Equal(tt.wantErr))
+			g.Expect(len(c.processMetrics()) > 0).To(Equal(tt.wantErr))
```

Comment:
Mind adding a comment to explain why `len(c.processMetrics()) > 0` is related to `tt.wantErr` ?

Reply:
That's perfect! Thank you!

Reply:
It is outside of the scope of this PR, but I was wondering if it wouldn't be better/more standard to return just one `error` containing a method (for example: `GetCause error[]`) that will return the list of errors.

Reply:
I think the idea behind return the error array was to able to iterate through all of them and capture them in sentry / logs.
This was driven by the `.Reconcile() []error` method signature. 
What will be the gain by encapsulating the errors in an error container with a `GetCause()`?

Reply:
Because when I invoke a method, I expect the `error` to be `nil` when no error has occurred, I don't expect it to be an empty array.

And when the error occurs I expect it to be a single object

Reply:
That's why I proposed an error with a `GetCause`  method. If no error is found, it would return `nil`. If an error is found it would return en error `Failure processing metrics` and the `GetCause` would contain the list of errors.

Reply:
About the `Log`, I would avoid that method: I prefer each object to have only one responsibility. We can however make a `SentryLogger` that, when a `CompoundError` is received, would logs each of the embedded errors.

Reply:
It's probably not worth creating another struct. We could extend the `ServiceError` to suit the multiple causes use case: https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/622303cad34fa07ba2aa29841104e6f239c24706/pkg/errors/errors.go#L312 which we've at the moment to allow taking in a list of error causes instead of just one: https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/622303cad34fa07ba2aa29841104e6f239c24706/pkg/errors/errors.go#L320

And it sees how everything fits together with the reconciler. As you mentioned, outside the scope of this PR but I am willing to chime in a followup discussion that will be opened.